src/core/layout/qgslayoutitemmap.cpp

```diff
@@ -1592,11 +1594,18 @@ void QgsLayoutItemMap::recreateCachedImageInBackground()
   if ( w <= 0 || h <= 0 )
     return;
 
-  mCacheRenderingImage.reset( new QImage( w, h, QImage::Format_ARGB32 ) );
+  double devicePixelRatio = 1.0;
+  if ( QWidget *activeWindow = QApplication::activeWindow() )
```

Comment:
Hmm, this is pretty ugly. Can't we get it from the paint device again?

Reply:
We can't from here, we haven't created the painter yet.

Reply:
but recreateCachedImageInBackground is always triggered via an initial call to paint -- so we could grab the painter ratio just before calling `mBackgroundUpdateTimer->start( 1 );` and store it for use in recreateCachedImageInBackground

Reply:
@nyalldawson , nice!